code/controllers/subsystem/air.dm

```diff
@@ -68,6 +71,25 @@ SUBSYSTEM_DEF(air)
 /datum/controller/subsystem/air/fire(resumed = 0)
 	var/timer = TICK_USAGE_REAL
 
+	if(pipenets_needing_rebuilt.len) // always do this nonsense first
+		var/oldpart = currentpart
+		src.currentrun = pipenets_needing_rebuilt.Copy()
+		//cache for sanic speed (lists are references anyways)
+		var/list/currentrun = src.currentrun
+		while(currentrun.len)
+			var/obj/machinery/atmospherics/thing = currentrun[currentrun.len]
+			currentrun.len--
+			if(thing && istype(thing, /obj/machinery/atmospherics)) // for some reason some pipenet datums keep making it into the queue and i'm not sure how, since its gated by an istype already
```

Comment:
This is an issue that should be solved.

Reply:
if its somehow breaking past a byond istype() i think its a byond bug

Reply:
```cpp
/datum/pipeline/New()
	SSair.networks += src
```
```
/datum/controller/subsystem/air/proc/process_pipenets(resumed = 0)
	if (!resumed)
		src.currentrun = networks.Copy()
```
Am I missing something here?

Reply:
oh god damn it that explains the problem

ima just yoink the atmos stuff for now I think tho